packages/labextension/src/index.ts

```diff
@@ -2,25 +2,37 @@ import {
   JupyterFrontEnd,
   JupyterFrontEndPlugin,
 } from '@jupyterlab/application';
+import { INotebookTracker } from '@jupyterlab/notebook';
+import { LabIcon } from '@jupyterlab/ui-components';
+import { ICommandPalette } from '@jupyterlab/apputils';
+import { KernelUsagePanel } from './panel';
+import tachometer from '../style/tachometer.svg';
 
 import { IStatusBar } from '@jupyterlab/statusbar';
 
 import { ITranslator } from '@jupyterlab/translation';
 
 import { MemoryUsage } from './memoryUsage';
 
+namespace CommandIDs {
+  export const getKernelUsage = 'kernel-usage:get';
+}
+
 /**
  * Initialization data for the jupyter-resource-usage extension.
  */
 const extension: JupyterFrontEndPlugin<void> = {
   id: '@jupyter-server/resource-usage:memory-status-item',
   autoStart: true,
-  requires: [IStatusBar, ITranslator],
+  requires: [IStatusBar, ITranslator, ICommandPalette, INotebookTracker],
```

Comment:
Usually `ICommandPalette` is consumed as `optional` to allow for more flexibility when reusing plugins elsewhere.

Reply:
Actually wondering if we could make most of these dependencies `optional`.

This could help make the extension work in Notebook 7 as well for example.

Reply:
Does it mean removing the `requires` entry altogether?

Reply:
I guess yes by just looking at the code. Or by splitting the functionalities in multiple plugins. 

Also the name of the plugin is still `@jupyter-server/resource-usage:memory-status-item'` but is providing more functionalities now.